core/src/providers/openai.rs

```diff
@@ -247,7 +249,7 @@ impl From<OpenAIChatMessageRole> for ChatMessageRole {
 #[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
 pub struct OpenAIChatMessage {
     pub role: OpenAIChatMessageRole,
-    pub content: Option<String>,
+    pub content: Option<ContentBlock>,
```

Comment:
This is not great to have external libraries types depend on our own type no?

Reply:
Let's have a local content with conversion traits?

Reply:
Implemented in d5a87a4333b78e6d66c4dbf58adc51196af69ab9.